frontend/components/Documents.js

```diff
@@ -0,0 +1,46 @@
+import React, {useEffect, useState} from "react";
+// import * as PropTypes from "prop-types";
+// import STYLES from "./Documents.module.scss";
+
+const Documents = () => {
+
+    const [docData, setDocData] = useState(null);
+
+    useEffect(() => {
+        fetch("/api/all_documents")
+            .then(response => response.json())
+            .then(data => {
+                setDocData(data);
+            });
+    }, []);
+
+    return (
+        <div>
+            <h1>This is the Documents page.</h1>
+            <p>
+                This page displays all the documents stored in backend.
+            </p>
+            {
+                docData
+                    ? <div>Documents:
+                        {docData.map((doc, i) => {
+                            return (<>
+                                <p>Document {i}</p>
+                                <ul key={i}>
+                                    {Object.keys(doc).map((key, i) => {
+                                        return (
+                                            <li key={i}>{key}: {doc[key]}</li>
+                                        );
+                                    })}
+                                </ul>
+                            </>
+                            );
+                        })}
+                    </div>
+                    : <p>Currently Loading Documents...</p>
+            }
```

Comment:
Another opinionated opinion: the `return` value of our components should be as clean and readable as possible. In this case, we have a significant amount of logic determining how this view is rendered. I'd recommend one of two things: 

1. pulling everything from line 25 to line 39 into a function inside the component:
```
const documentList = () => {
        return (
            <div>
                Documents:
                { etc.
```
And then calling that in line 25, or

2. Creating a new React component to encapsulate this view and passing `docData` into it, then referencing that component in line 25. 

In either case, doing so would allow subsequent developers to more quickly skim the return value and understand, generally, what's being rendered, and, should they want to dive into the logic, finding it in the appropriate place. Given the verbose nature of HTML, skimming a lengthy set of logic like this to understand what is likely to be rendered on the page can become quite challenging.

Reply:
I've moved the code into two separate functions `docList` and `docInfo` to allow for easier updates in the future if we decide to change the way we display the document information.